site/content/blog/incorporating-external-resources-in-kapp.md

```diff
@@ -0,0 +1,161 @@
+---
+title: "Incorporating external resources in kapp"
+slug: incorporating-external-resources-in-kapp
+date: 2021-12-24
+author: Praveen Rewar
+excerpt: "How to use kapp's exists annotation to wait for resources created by external agencies"
+image: /img/logo.svg
+tags: ['Praveen', 'kapp', 'exists']
+---
+
+kapp CLI encourages Kubernetes users to manage resources in bulk by working with "Kubernetes applications" (sets of resources with the same label). But there are often times when we want to incorporate resources that are not actually part of the same application (created by external agents). 
+
+In this blog, we are going to learn how to use the `kapp.k14s.io/exists` annotation to wait for resources that are not owned by kapp.
+
+## Problem Statement
+Let's take a look at an example application which consists of a _configmap_ and a _secret_ both of which needs to be created in a namespace which is not part of the application and it will be created by an external agency (such as a controller).
```

Comment:
For more clarity, we can break into two lines. This will help in emphasising that namespace is not part of the same application

Reply:
Sure!